### PR TITLE
Scheduler implementation

### PR DESCRIPTION
**Pull request by Enlarem**
Added a new page with the scheduler. Everything related to event modification and deletion should be working.
I modified, and added the following behaviors : 

- In EventItem.js , when the function "onDeleteEvent" is defined in the props, it is triggered when the current event is deleted
- The Event Item now updates when the props does
- The Ticket data fetching is made again when the props are updating

The rest should be located only in the Scheduler page file

### CODE REVIEW

**Comment by Jezerig** [comment]
image
For some reason it now doesn't display any events because of the thing I added

**Comment by Enlarem** [comment]
It should be working again now

**Comment by Jezerig** [comment]
- If I edit an event ending time for example, the calendar doesn't update until I refresh.
- After I edited an ending time of an event, if displayed that ending time for all events when I click edit on any of them
- Honestly, the whole event editing is such a mess I don't know which ones are only on this branch and which ones are also in Main

**Comment by Enlarem** [comment]
I think I solved the refreshing problems related to editing the event. For that, I added a new optional "callback" prop that is triggered when the event is being saved.

File: client/src/components/EventItem.js [hunk 1]
```diff
@@ -442,26 +465,6 @@ const handleJoinDeadlineError = (error) => {
...
-        {
-          setLiking(data.attending);
-          setLoadingLikes(false);
-        }
-      );
-    }
-
-    // Find out if the user is like this event or not
-    confirmLike();
-  }, []);
```
Comment (Enlarem) [inline client/src/components/EventItem.js:486]: Also, I moved this piece of code in the above useEffect triggered when the props are changed. It results on the same effects, and the like button is no well updated

**Review by jannejjj — COMMENTED** [review]
Already mentioned this on Discord, but multi-day events that span over a Saturday don't show up correctly, it shows that there's nothing on Saturday for some reason.. This might be a bug with the actual calendar package, so idk if there's that much we can do about it
image
In this photo, the Lahti excursion spans Thu-Sun, but isn't shown on Saturday

+the use button on tickets is a bit too far to the right atl east with my current resolution (1920x1080)

image

**Review by Jezerig — APPROVED** [review]
- Event that spans over Saturday works for me dunno why
- Functionality wise everything seems to work now, Good job! Just make the changes Janne requested

File: client/src/components/EventItem.js [hunk 2]
```diff
@@ -77,30 +77,40 @@ function EventItem(props) {
...
+    setTitle(props.event.title);
+    setLocation(props.event.location);
+    setDescription(props.event.description);
+    setTicketsSold(props.event.ticketsSold);
+    setTickets(props.event.tickets);
+    setPrice(props.event.price);
+
+  }, [props.event]);
+
+  // useEffect(() => { // Get event participants
```
[the remark this answers is not in the document]
Comment (Enlarem) [reply client/src/components/EventItem.js]: It was a duplicate of the useEffect called just above I think

**Review by jannejjj — APPROVED** [review]
I still have the _saturday issue_ but since it seems like you guys don't, and it's not raelly something we could fix anyway, everything looks good now!

**Pull request merged**